CapstoneProject/Assets/Scripts/Core/Controller/PlayerController.cs

```diff
@@ -47,7 +61,19 @@ private void Update()
                 if (IsPressed(PlayerInputReader.Instance.attackValue))
                     _playerStateComponent.stateMachine.ChangeState(_playerStateComponent.groundAttackState);
                 if (IsPressed(PlayerInputReader.Instance.dashValue))
-                    _playerStateComponent.stateMachine.ChangeState(_playerStateComponent.dashState);
+                {
+                    if (_playerAbilityComponent.DashAbility.CanUseAbility())
+                    {
+                        _playerStateComponent.stateMachine.ChangeState(_playerStateComponent.dashState);
+                    }
+                }
+                if (IsPressed(PlayerInputReader.Instance.firstAbilityValue) && _playerAbilityComponent.playerAbilities[0] != null)
+                {
+                    if (_playerAbilityComponent.playerAbilities[0].CanUseAbility())
```

Comment:
could we not just add this check directly inside the ExecuteAbilities function so that if you need to fix it for one ability it fixes for all and reduces the code slightly?